xDrip/Model/Entries/TrainingEntry.swift

```diff
@@ -9,9 +9,33 @@
 import Foundation
 import RealmSwift
 
-final class TrainingEntry: AbstractEntry {
+final class TrainingEntry: AbstractEntry, TreatmentEntryProtocol {
     @objc private(set) dynamic var duration: TimeInterval = 0.0
     @objc private dynamic var rawIntensity: Int = TrainingIntensity.default.rawValue
+    @objc private(set) dynamic var externalID: String?
+    @objc private dynamic var rawCloudUploadStatus: Int = CloudUploadStatus.notApplicable.rawValue
+    
+    var cloudUploadStatus: CloudUploadStatus {
+        get {
+            return CloudUploadStatus(rawValue: rawCloudUploadStatus) ?? .notApplicable
+        }
+        set {
+            rawCloudUploadStatus = newValue.rawValue
+        }
+    }
+    
+    internal var amount: Double {
+        return duration
+    }
+    
+    var exerciseIntensity: Int? {
+        get {
+            return rawIntensity
+        }
+        set {
+            rawIntensity = newValue ?? 1
```

Comment:
You're exposing Realm setter outside of the class. If it will be called outside of Realm.write closure, it will crash.